ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/avdtestbase.py

```diff
@@ -43,6 +43,162 @@ def render(self) -> dict:
         """
         return getattr(self, "test_definition", None) or {}
 
+    def log_message(
+        self, message=None, key: str | None = None, value=None, key_path: str | None = None, is_missing: bool = True, logging_level: str = "INFO"
+    ) -> None:
+        """
+        Flexible logging function that logs a formatted message based on the provided parameters.
+
+        Parameters:
+            message (Any | None): The message to be logged. If provided, it will be logged as is, ignoring other parameters.
+            key (str | None): The key to be logged.
+            value (Any | None): The expected value of the key. Must be provided when logging an invalid value.
+            key_path (str | None): The key path in dot notation.
+            is_missing (bool): Indicates whether the key is missing.
+            logging_level (str): The logging level to use for the log message.
+        """
+        logging_levels = ["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]
+
+        # Validate logging level
+        if logging_level.upper() not in logging_levels:
+            raise AristaAvdError("Invalid logging level. Please choose from DEBUG, INFO, WARNING, ERROR, CRITICAL.")
+
+        # If message is provided, it will be logged as is, ignoring other parameters
+        if message:
+            log_msg = str(message)
+        else:
+            if key is None:
+                raise AristaAvdError("Error creating the log message: Key argument is missing.")
+            if value is None and is_missing is False:
+                raise AristaAvdError("Error creating the log message: The key's value must be provided when logging an invalid value.")
+            dot_notation = f"{key_path}.{key}" if key_path else f"{key}"
+            msg_type = "is missing" if is_missing else f"!= '{value}'"
+            log_msg = f"Key '{dot_notation}' {msg_type}."
+
+        log_msg += f" {self.__class__.__name__} is skipped."
```

Comment:
So this logging method _always_ logs that the test is skipped. That is not clear from the doc string, and may not be what you really want? If it is, maybe call it `log_skip_message`

Reply:
Fixed in 7c1821824bc682cfd9015747f8841e2e8f84991f